sdks/python/apache_beam/ml/rag/embeddings/huggingface.py

```diff
@@ -0,0 +1,76 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""RAG-specific embedding implementations using HuggingFace models."""
+
+from typing import Optional
+
+import apache_beam as beam
+from apache_beam.ml.inference.base import RunInference
+from apache_beam.ml.rag.embeddings.base import create_rag_adapter
+from apache_beam.ml.rag.types import Chunk
+from apache_beam.ml.rag.types import Embedding
+from apache_beam.ml.transforms.base import EmbeddingsManager
+from apache_beam.ml.transforms.base import _TextEmbeddingHandler
+from apache_beam.ml.transforms.embeddings.huggingface import SentenceTransformer
+from apache_beam.ml.transforms.embeddings.huggingface import _SentenceTransformerModelHandler
+
+
+class HuggingfaceTextEmbeddings(EmbeddingsManager):
+  """SentenceTransformer embeddings for RAG pipeline.
+    
+    Extends EmbeddingsManager to work with RAG-specific types:
+    - Input: Chunk objects containing text to embed
+    - Output: Embedding objects containing vector representations
+    
+    The adapter automatically:
+    - Extracts text from Chunk.content.text
+    - Preserves Chunk.id in Embedding.id
+    - Copies Chunk.metadata to Embedding.metadata
+    - Converts model output to Embedding.dense_embedding
```

Comment:
Could we also follow this convention for parameters for pydocs - https://github.com/apache/beam/blob/a6061feb05363a6175e28f80f684b6d306a42150/sdks/python/apache_beam/ml/transforms/base.py#L247

That will allow them to be picked up as known params in the rendered doc